### PR TITLE
feat: support react 18

### LINKED ISSUE #893: Incompatible with react v18

### PR DESCRIPTION
**Pull request by redonkulus**
Fixes #893

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: tests/unit/Sticky.test.js [hunk 1]
```diff
@@ -395,28 +445,38 @@ describe('Sticky', () => {
...
+            window.scrollTo(0, 1500);
+        });
         shouldBeFixedAt(inner, -432);
         expect(outer.className).toContain('active');
         expect(outer.className).not.toContain('released');
 
+        // !!!! THESE TESTS FAIL, NEED TO FIGURE OUT WHY !!!!
+
         // Change Sticky's height
         STICKY_HEIGHT = 1300;
```
[the remark this answers is not in the document]
Comment (AnthonyCrowcroft) [reply tests/unit/Sticky.test.js:466]: was it required to update the test utils?

[the remark this answers is not in the document]
Comment (AnthonyCrowcroft) [reply tests/unit/Sticky.test.js:466]: Upon looking into the dependencies it was required to update the testing utils, sorry for the dumb question. 

When comparing the single failing test to the rest the only difference I could immediately find was that it changed the STICKY_HEIGHT after the render and it doesn't seem to have changed the value as all expect checks post that are not passing.

@redonkulus any ideas on next steps?

[the remark this answers is not in the document]
Comment (redonkulus) [reply tests/unit/Sticky.test.js:466]: Really just debugging further into why the height value is t coming back, I assume it’s some internal React data structure change that we need to change to.

**Comment by redonkulus** [comment]
I'm going to merge this so users can use this lib with latest React, I will open an issue to track the failing test.

**Pull request merged**